dali/python/nvidia/dali/plugin/paddle.py

```diff
@@ -72,6 +73,8 @@ def feed_ndarray(dali_tensor, ptr, cuda_stream=None):
                     (if not provided, an internal user stream will be selected)
     """
 
+    if cuda_stream is None:
```

Comment:
How about moving this to L326 (like in the case of pytorch)?
Please also remove `fluid.core._cuda_synchronize(pd_gpu_place)` which was supposed to be a workaround for the mentioned problem.
Side question - which paddle paddle version is needed to support this maybe there should an if for the version?
And you can update qa/setup_packages.py to make DALI tests use the more recent paddlepaddle-gpu version.

Reply:
Done. Thanks for your suggestions!